### ISSUE TITLE
Autocomplete the html tags when hitting enter

### ISSUE THREAD

**Issue by thewhitetulip** [adobe/brackets#10799]
Currently we have to do a `<ul` then the autocomplete dialog comes, hit enter, then type `>` and then the tag is completed, it would be cool to type `<ul` -- autocomplete dialog -- hit enter and it automatically adds the tags.

**Comment by nethip** [comment]
@thewhitetulip Good idea! I believe this is an excellent candidate for this being an extension. I need to check for technical feasibility if we have infrastructure to support this as an extension. For now I am going to add "extension" tag to this.

**Comment by thewhitetulip** [comment]
Okay, but I thought it would be better off as being built right into brackets as opposed to an extension. I am not a big fan of overly extensible editors like Atom where everything is an extension!

**Comment by nethip** [comment]
@thewhitetulip The way extension work in Brackets is very different to that of other apps. Technically speaking, usually you would find a "glue layer" that takes care of communication between the client and the extension in other apps. In these scenarios extension becomes a second class citizen. 

But in case of Brackets this gluing is minimal, meaning once an extension is loaded, it becomes part of the client. That's the beauty of Brackets! Keeping aside the technical jargon, writing a new extension in Brackets is similar to writing a new feature in Brackets. In fact we ship many of our features as extensions!. Checkout the source for them at https://github.com/adobe/brackets/tree/master/src/extensions/default.

**Comment by thewhitetulip** [comment]
That is an excellent architecture then! Normally extensions are built on top of the core product, impressive :+1:

**Comment by Wikunia** [comment]
Hey @thewhitetulip maybe you want to checkout this extension:
https://github.com/emmetio/brackets-emmet

**Comment by thewhitetulip** [comment]
Hey @Wikunia, yep, I came across that extension! would it be possible to make it a default package?

**Comment by bjornpaulstrom** [comment]
Me and some friends are working on this for an university course, the current implementation is located here:

https://github.com/Matlal/brackets-HTML-Tag-Autocomplete

It supports both self closing tags and normal tags.

**Comment by bjornpaulstrom** [comment]
The extension is now uploaded to the Brackets extension library :)

**Comment by myasul** [comment]
@bjornpaulstrom I tried it out. It works like magic. Two thumbs up! Thanks man!
